src/main/java/com/google/sps/servlets/RemoveFromGrocery.java

```diff
@@ -26,21 +26,17 @@
  
 public class RemoveFromGrocery {
  
-public void removeMember(String removedMember) {
-    UserService userService = UserServiceFactory.getUserService();
+  public void removeMember(String removedMember) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Key groceryEntityKey = KeyFactory.createKey("Grocery", removedMember);
     
-     Entity groceryEntity;
-    try {
-        groceryEntity = datastore.get(groceryEntityKey);
-    } catch (EntityNotFoundException e) {
-        System.out.println("Member not found");
-        return;
+    Query removedMemberQuery = new Query(GROCERY)
+      .setFilter(new Query.FilterPredicate(“memberEmail”, Query.FilterOperator.EQUAL, removedMember)); 
+
+    PreparedQuery results = datastore.prepare(removedMemberQuery);
+
+    for (Entity memberEntity : results.asIterable()) {
```

Comment:
change memberEntity to groceryEntity maybe because this is looping through grocery items